internal/pkg/build/assemblers/sif.go

```diff
@@ -177,7 +179,12 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) error {
 	if a.GzipFlag {
 		flags = append(flags, "-comp", "gzip")
 	}
-
+	if a.MksquashfsMem != "" {
+		flags = append(flags, "-mem", a.MksquashfsMem)
```

Comment:
Looks like `-mem` is not portable enough, old `mksquashfs` version doesn't have this option

Reply:
From the changelog it appears that this option was added to mksquashfs in May of 2014.  I think this is a useful configurable option (I know I need it).  Any ideas as to how to get this available without breaking people?  I can change it so that if the value is not set in the config file it doesn't pass the command line argument at all and add a note about version needed in the singularity.conf file.  Could we add a version requirement for mksquasfs to ensure that it is a new enough version that it is supported?  Perhaps, if not, add it as a configurable value for mconfig that will be required to enable it to show in the config file?  Other ideas?

Reply:
IIRC, RHEL 6 and SLES 11 don't have a new enough version. RHEL 6 is EOL 2024, and though SLES 11 is EOL now... it's still in use quite a bit.

Reply:
Sounds reasonable, if you could add a note for `mksquashfs mem` directive in template ? to mention that it may not work with older version of mksquashfs and in this case it must be left blank/empty